File: Array/liuchang/day23/day23.cpp
//
// Created by 刘畅 on 2020/6/23.
//

#include "day23.h"
#include "iostream"
#include "math.h"
int findNumbers(int* nums, int numsSize);

int findNumbers(int* nums, int numsSize){
    int count = 0;
    for (int i = 0; i <numsSize ; ++i) {
        int n  = 1;
        while (pow(10,n)<nums[i]){
            n++;
        }
        if (n%2 ==0){
            count++;
        }
    }
    return count;
}

int main(){
    int nums[] = {12,345,2,6,7896};
    printf("%d\n",findNumbers(nums,5));
    return 0;
}

File: Array/liuchang/day23/day23.java
import java.lang.Math;

public class day23 {

    public static void main(String[] args) {
        int nums[] = {12, 345, 2, 6, 7896};
        System.out.println("=====" + findNumbers(nums, 5));
    }

    public static int findNumbers(int data[], int size) {
        int count = 0;
        for (int i = 0; i < size; ++i) {
            int n = 1;
            while (Math.pow(10, n) < data[i]) {
                n++;
            }
            if (n % 2 == 0) {
                count++;
            }
        }
        return count;
    }
}
